### DISCUSSION TITLE
add new firmware for Bluetooth dongle

### DISCUSSION THREAD

**Discussion by viotemp1** [home-assistant/operating-system#2079] [Q&A]
Hello,

How can I add new firmware files for my BT dongle (LM1010-0972)?

https://github.com/winterheart/broadcom-bt-firmware/raw/master/brcm/BCM20702A1-0a5c-21e8.hcd

[    3.212251] NET: Registered PF_BLUETOOTH protocol family
[    3.212262] Bluetooth: HCI device and connection manager initialized
[    3.212284] Bluetooth: HCI socket layer initialized
[    3.212301] Bluetooth: L2CAP socket layer initialized
[    3.212322] Bluetooth: SCO socket layer initialized
[    3.280690] usbcore: registered new interface driver btusb
[    3.402194] Bluetooth: hci0: BCM: chip id 63
[    3.403506] rpivid-mem feb00000.hevc-decoder: rpivid-hevcmem initialised: Registers at 0xfeb00000 length 0x00010000
[    3.403909] rpivid-mem feb10000.rpivid-local-intc: rpivid-intcmem initialised: Registers at 0xfeb10000 length 0x00001000
[    3.405165] Bluetooth: hci0: BCM: features 0x07
[    3.415249] rpivid-mem feb20000.h264-decoder: rpivid-h264mem initialised: Registers at 0xfeb20000 length 0x00010000
[    3.423195] Bluetooth: hci0: BCM20702A
[    3.423238] Bluetooth: hci0: BCM20702A1 (001.002.014) build 0000
[    3.427814] Bluetooth: hci0: BCM: firmware Patch file not found, tried:
[    3.427849] Bluetooth: hci0: BCM: 'brcm/BCM20702A1-0a5c-21e8.hcd'
[    3.427860] Bluetooth: hci0: BCM: 'brcm/BCM-0a5c-21e8.hcd'
Regards,

**Comment by davidandreoletti** [comment]
@viotemp1 The [HA bluetooth integration](https://www.home-assistant.io/integrations/bluetooth/#broadcom-bcm-based-adapters) states one can install drivers from https://github.com/winterheart/broadcom-bt-firmware. 

However, the HA OS's /lib/firmware path is mounted as overlayfs and at reboot time, the newly installed the Broadcom firmware will be gone from this path. Back to square one :-(

@sairon 
1. How to add the broadcom firmware with overlayfs ? 
2. Why not integrate the firmware within the image itself ?
